lib/vanagon/driver.rb

```diff
@@ -27,17 +28,35 @@ def initialize(platform, project, options = { :configdir => nil, :target => nil,
       @project.settings[:skipcheck] = options[:skipcheck]
       loginit('vanagon_hosts.log')
 
-      # If a target has been given, we don't want to make any assumptions about how to tear it down.
-      engine = 'base' if target
-      # Hardware has explicit teardown to unlock the node
-      engine = 'hardware' if @platform.build_hosts
-      require "vanagon/engine/#{engine}"
-      @engine = Object.const_get("Vanagon::Engine::#{engine.capitalize}").new(@platform, target)
-
+      load_engine(engine, @platform, target)
     rescue LoadError => e
       raise Vanagon::Error.wrap(e, "Could not load the desired engine '#{engine}'")
     end
 
+    def load_engine(engine_type, platform, target)
+      if platform.build_hosts
+        engine_type = 'hardware'
+      elsif target
+        engine_type = 'base'
+      end
+      load_engine_object(engine_type, platform, target)
+    end
+
+    def load_engine_object(engine_type, platform, target)
+      case engine_type
+      when 'hardware'
+        @engine = Vanagon::Engine::Hardware.new(platform, target)
+      when 'docker'
+        @engine = Vanagon::Engine::Docker.new(platform, target)
+      when 'local'
+        @engine = Vanagon::Engine::Local.new(platform, target)
+      when 'pooler'
+        @engine = Vanagon::Engine::Pooler.new(platform, target)
+      else
+        @engine = Vanagon::Engine::Base.new(platform, target)
+      end
```

Comment:
This could probably all be collapsed into 

```ruby
def load_engine_object(engine_type, platform, target)
  Object::const_get("Vanagon::Engine::#{engine_type.capitalize}").new(platform, target)
rescue
  fail "No such engine '#{engine_type.capitalize}'"
end
```